binary_search/binary_search.c

```diff
@@ -5,6 +5,37 @@
  * @arr_size Size of array
  * @search_element Element to be searched
  */
+ 
+ /* function to perform recursive binary search
+ input parameters:
+ ar - user-input array.
+ l - left index of array.
+ r - right index of array.
+ ele - element to be searched.
+ */
+ int binarySearch(const int *ar, int l, int r, int ele)
```

Comment:
name of this function should be `binary_search_recursive` to show that it is a recursive solution and also to follow guidelines for cpp.